nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js

```diff
@@ -0,0 +1,31 @@
+const express = require("express");
+const router = express.Router();
+
+const reservations = require("../data/reservations.json");
+
+router.get("/", async (request, response) => {
+    try {
+        console.log(reservations);
+        console.log("in /api/reservations");
+        response.send(reservations);
+    } catch (error) {
+        throw error;
+    }
+});
+router.get("/:id", async (request, response) => {
+
+    if (typeof parseInt(request.params.id) == "number") {
+        const responseData = reservations.filter(reservation => reservation.id === parseInt(request.params.id))
+        if (responseData.length === 0) {
+            return res.status(404).send("Given query does not find any data")
+        }
+        else {
+            return response.send(responseData);
+        }
+    }
+    else {
+        return res.status(400).send("Given parameter is not supported")
```

Comment:
'Res' is not defined. Should be changed to 'response', since this is what you wrote in line 6.

Reply:
fixed